src/main/kotlin/io/libp2p/pubsub/gossip/GossipRouter.kt

```diff
@@ -173,7 +174,28 @@ open class GossipRouter @JvmOverloads constructor(
     }
 
     override fun acceptRequestsFrom(peer: PeerHandler): Boolean {
-        return isDirect(peer) || score.score(peer) >= score.params.graylistThreshold
+        if (isDirect(peer)) {
+            return true
+        }
+
+        val curTime = curTimeMillis()
+        val whitelistEntry = acceptRequestsWhitelist[peer]
+        if (whitelistEntry != null &&
+            curTime <= whitelistEntry.whitelistedTill &&
+            whitelistEntry.messagesAccepted < acceptRequestsWhitelistMaxMessages
+        ) {
+
+            acceptRequestsWhitelist[peer] = whitelistEntry.incrementMessageCount()
+            return true
+        }
+
+        val peerScore = score.score(peer)
+        if (peerScore >= acceptRequestsWhitelistThresholdScore) {
+            acceptRequestsWhitelist[peer] =
+                AcceptRequestsWhitelistEntry(curTime + acceptRequestsWhitelistDuration.toMillis())
+        }
```

Comment:
Should we clear out the whitelist entry otherwise?

Reply:
Yep, makes sense 👍 
c545f19